modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/message/body/BaseMessageBodyReader.java

```diff
@@ -94,6 +152,35 @@ private ObjectMapper _getObjectMapper(Class<?> clazz) {
 		);
 	}
 
+	private void _setExtendedProperties(
+			Class<?> clazz, ExtensionContext extensionContext,
+			JsonNode jsonNode, Object object, ObjectMapper objectMapper)
+		throws IOException {
+
+		List<String> clazzFieldNames = new ArrayList<>();
+
+		for (Field field : clazz.getDeclaredFields()) {
+			clazzFieldNames.add(field.getName());
+		}
+
+		Iterator<String> fieldNamesIterator = jsonNode.fieldNames();
+		Map<String, Object> extendedProperties = new HashMap<>();
+
+		while (fieldNamesIterator.hasNext()) {
+			String jsonFieldName = fieldNamesIterator.next();
+
+			if (!clazzFieldNames.contains(jsonFieldName)) {
+				JsonNode fieldNameJsonNode = jsonNode.get(jsonFieldName);
+
+				extendedProperties.put(
+					jsonFieldName,
+					_getJsonNodeValue(fieldNameJsonNode, objectMapper));
+			}
+		}
+
+		extensionContext.setExtendedProperties(object, extendedProperties);
```

Comment:
Is necessary to call `setExtendedProperties` even if there are no properties to set?

Reply:
Good point.

It was unintentional but thinking about it now, maybe it is a good idea to allow setting default values or making some checks.

I would leave it like that and maybe clarify in the usage that it is meant for this purpose.

Reply:
I like the idea. One question, where is documented the usage? In the Pull request / Jira issue?

Reply:
We will need to document it after the final version is merged. Should be one task of the epic (the last one) and in this kind of epic is very important to document it as would be an "important" feature